### PR TITLE
[Yul] Use map join algorithm for performance.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

**Comment by leonardoalt** [comment]
Test failing

File: test/libyul/yulOptimizerTests/fullSuite/abi_example1.yul [hunk 1]
```diff
@@ -0,0 +1,623 @@
+{
```
Comment (leonardoalt) [inline test/libyul/yulOptimizerTests/fullSuite/abi_example1.yul:1]: Not really sure how to review this file...

Comment (chriseth) [reply test/libyul/yulOptimizerTests/fullSuite/abi_example1.yul:1]: The point of `fullSuite` is not necessarily correctness (that is tested in the individual steps) but rather checking that the result does not miss any optimizations.

**Comment by chriseth** [comment]
Updated.

**Comment by chriseth** [comment]
Updated.

File: libyul/optimiser/RedundantAssignEliminator.cpp [hunk 2]
```diff
@@ -150,30 +156,48 @@ void RedundantAssignEliminator::run(Block& _ast)
...
 	remover(_ast);
 }
 
-void RedundantAssignEliminator::join(RedundantAssignEliminator& _other)
+template <class K, class V, class F>
+void joinMap(std::map<K, V>& _a, std::map<K, V>&& _b, F _conflictSolver)
 {
-	for (auto& var: _other.m_assignments)
-		if (m_assignments.count(var.first))
+	// TODO Perhaps it is better to just created a sorted list
```
Comment (leonardoalt) [inline libyul/optimiser/RedundantAssignEliminator.cpp]: Small typo here `created` -> `create`

**Review by leonardoalt — DISMISSED** [review]

**Pull request merged**